src/malli/core.cljc

```diff
@@ -275,6 +275,47 @@
           (-properties [_] properties)
           (-form [_] form))))))
 
+(defn- -map-of-schema []
+  ^{:type ::into-schema}
+  (reify IntoSchema
+    (-name [_] :map-of)
+    (-into-schema [_ properties childs opts]
+      (when-not (and (seq childs) (= 2 (count childs)))
+        (fail! ::invalid-map-of))
+      (let [[key-schema value-schema :as schemas] (mapv #(schema % opts) childs)
+            key-valid? (-validator key-schema)
+            value-valid? (-validator value-schema)
+            validate (fn [m]
+                       (reduce
```

Comment:
`reduce-kv` should be used

Reply:
also guard that the value is a `map?`

Reply:
Are you talking about this? https://github.com/metosin/malli/pull/41/files#diff-80b352d39081ba397dc83fc6fafe7a47R294

Reply:
Oh, the `map?` was there already, my bad